pkg/vm/create_evm.go

```diff
@@ -118,8 +120,17 @@ func CreateEVMGenesis(
 		}
 	}
 
+	if params.UsePoAValidatorManager && params.UsePoSValidatorManager {
+		return nil, fmt.Errorf("blockchain can not be both PoA and PoS")
+	}
 	if params.UsePoAValidatorManager {
 		validatormanager.AddPoAValidatorManagerContractToAllocations(params.initialTokenAllocation)
+		validatormanager.AddTransparentProxyContractToAllocations(params.initialTokenAllocation, proxyOwner)
+	} else {
+		validatormanager.AddPoSValidatorManagerContractToAllocations(params.initialTokenAllocation)
```

Comment:
can it be this case that it is neither poa neither pos (old stuff)

Reply:
good catch. replaced with `else if`